### PR TITLE
added index for sender

### PR DESCRIPTION
**Pull request by maikelmclauflin**
subtle bug for sender address derivation

### CODE REVIEW

**Comment by reidrankin** [comment]
(I've picked up this fix in #376)

**Review by davidp94 — APPROVED** [review]

**Review by veado — COMMENTED** [review]
Not sure if I miss something, but it seems this PR just added a `try/catch` block, we don't want anymore. Changes of 57de816fc82e54cb82988a03258466f4af7d3cfb seems to be already in `master`. In this case, let's close this PR.

**Review by veado — CHANGES_REQUESTED** [review]
https://github.com/xchainjs/xchainjs-lib/pull/375#pullrequestreview-700972457

**Comment by davidp94** [comment]
> [#375 (review)](https://github.com/xchainjs/xchainjs-lib/pull/375#pullrequestreview-700972457)

I don't see it there https://github.com/xchainjs/xchainjs-lib/blob/master/packages/xchain-bitcoincash/src/client.ts#L347

**Comment by davidp94** [comment]
I will remove the try catch block

File: packages/xchain-bitcoincash/src/client.ts [hunk 1]
```diff
@@ -358,7 +358,7 @@ class Client extends BaseXChainClient implements BitcoinCashClient, XChainClient
     const tx = builder.build()
     const txHex = tx.toHex()
 
-    return await broadcastTx({
+    return broadcastTx({
```
Comment (reidrankin) [inline packages/xchain-bitcoincash/src/client.ts:361]: They're functionally equivalent, but `return await` gives nicer stack traces.

**Review by reidrankin — COMMENTED** [review]
I'll merge this this morning after I get #386 in.

**Comment by davidp94** [comment]
Thanks.

Le mer. 7 juil. 2021 à 13:39, Reid Rankin ***@***.***> a
écrit :

File: packages/xchain-bitcoincash/src/client.ts (hunk 1 above)
Comment (reidrankin) [reply packages/xchain-bitcoincash/src/client.ts:361]:
```suggestion
    return await broadcastTx({
```

**Review by veado — APPROVED** [review]

**Review by reidrankin — APPROVED** [review]
rebased and squashed

**Pull request merged**